File: python/SivitasAkademik.py
from Human import Human

class SivitasAkademik(Human):
    # Inisialisasi variabel untuk asal universitas dan email edu
    _asal_universitas = ""
    _email_edu = ""

    # Konstruktor untuk menginisialisasi variabel NIK, nama, jenis kelamin, asal universitas, dan email edu
    def __init__(self, nik='', nama='', jenis_kelamin='', asal_universitas='', email_edu=''):
        super().__init__(nik, nama, jenis_kelamin)
        self._asal_universitas = asal_universitas
        self._email_edu = email_edu

    # Getter untuk mendapatkan nilai asal universitas
    def getAsalUniversitas(self):
        return self._asal_universitas

    # Getter untuk mendapatkan nilai email edu
    def getEmailEdu(self):
        return self._email_edu

File: cpp/SivitasAkademik.cpp
// // import library
// #include <iostream>
// #include <string>

// using namespace std; //menggunakan namespace std

// #include "Human.cpp" //mengimpor file Human.cpp

class SivitasAkademik : public Human { //membuat kelas SivitasAkademik yang merupakan turunan dari kelas Mahasiswa
    protected: //mendefinisikan akses proteksi
        string asalUniversitas; //mendeklarasikan variabel asalUniversitas bertipe string
        string emailEdu; //mendeklarasikan variabel emailEdu bertipe string

    public: //mendefinisikan akses publik

        SivitasAkademik(string nik = "", string nama = "", string jenisKelamin = "", string asalUniversitas = "", string emailEdu = "") : 
        Human(nik, nama, jenisKelamin){ //mendefinisikan konstruktor dengan parameter untuk kelas SivitasAkademik yang juga memanggil konstruktor dengan parameter dari kelas Mahasiswa
            this->asalUniversitas = asalUniversitas; //menginisialisasi variabel asalUniversitas dengan nilai yang diterima melalui parameter
            this->emailEdu = emailEdu; //menginisialisasi variabel emailEdu dengan nilai yang diterima melalui parameter
        }

        string getAsalUniversitas(){return asalUniversitas;}
        string getEmailEdu(){return emailEdu;}

        ~SivitasAkademik(){} //mendefinisikan destruktor untuk kelas SivitasAkademik
};